### DISCUSSION TITLE
I have one issue adding formly inside material tabs-> mat-stepper

### DISCUSSION THREAD

**Discussion by shinde-prasad** [ngx-formly/ngx-formly#2769] [Q&A]
https://stackblitz.com/edit/angular-apzubb?file=src%2Fapp%2Fapp.component.html

here is sample issue i have created
if you check formly is not behaving seperate for two tabs, if tab1 form is valid tab2 form also get valid even if its values are not filled.

how to solve this?

@aitboudad #2633 #2548 #1885

**Comment by aitboudad** [comment]
sorry I'm unable to reproduce it, not sure where did you get the example but seems outdated see https://github.com/ngx-formly/ngx-formly/issues/2596#issuecomment-753509989

**Reply by shinde-prasad** [reply]
can you please refer the stackblitz. let me know that works for you.

**Reply by shinde-prasad** [reply]
as per your extras: { immutable: true } solution 
<formly-form [model]="model" (modelChange)="model = $event"`

I did this but not able to get values in eventemitter which I am using to expose values to <my-custom-compoent>.

any help on this.

**Reply by aitboudad** [reply]
Tested with the provided `stackblitz` and it seems working:

![image](https://user-images.githubusercontent.com/1753742/112690169-a416a080-8e7b-11eb-97b4-66ed81a4b4e5.png)

**Reply by shinde-prasad** [reply]
yes for stackblitz its working but on my application for <custom-component> its not working as [model]=[ngxformly] this doesnt take updated values.
